### PR TITLE
mozjs52: update to 52.9.0.

### PR DESCRIPTION
**Pull request by Cogitri**

### CODE REVIEW

File: srcpkgs/mozjs52/template [hunk 1]
```diff
@@ -12,7 +12,7 @@ homepage="http://www.mozilla.org/js/"
 license="MPL-1.1, GPL-2, LGPL-2.1"
```
Comment (maxice8) [inline srcpkgs/mozjs52/template]: please use SPDX short-identifiers

**Comment by Cogitri** [comment]
This also enables jemalloc for glibc builds, since jemalloc's performance is still better than glibc's malloc. Should be tested once icu rebuilding has been done.

**Comment by maxice8** [comment]
You will need this patch, put it in the patches subdirectory with a .patch prefix.

```
--- memory/mozjemalloc/jemalloc.c	2018-09-07 01:30:40.056777558 -0300
+++ memory/mozjemalloc/jemalloc.c	2018-09-07 01:30:45.847777478 -0300
@@ -331,9 +331,6 @@
 #endif
 #include <sys/time.h>
 #include <sys/types.h>
-#if !defined(MOZ_MEMORY_SOLARIS) && !defined(MOZ_MEMORY_ANDROID)
-#include <sys/sysctl.h>
-#endif
 #include <sys/uio.h>
 #ifndef MOZ_MEMORY
 #include <sys/ktrace.h> /* Must come after several other sys/ includes. */
```

**Comment by Cogitri** [comment]
> Are there any (real world) benchmarks that support that statement?

I can't seem to find benchmarks for spidermonkey with and without jemalloc right now, but [other](https://github.com/rust-lang/rust/issues/6897) [packages](http://engineering.appfolio.com/appfolio-engineering/2018/2/1/benchmarking-rubys-heap-malloc-tcmalloc-jemalloc) do have rather big performance improvements.

Also, Mozilla [has been using jemalloc by default for years now](https://bugzilla.mozilla.org/show_bug.cgi?id=1134039), so I feel like it's a good idea if we use it as well.

> I don't see where mozjs52 is used in performance critical packages.

Hm, yeah, once gjs 1.54.0 is in the tree the only packages consuming it seem to be cjs,goffie&polkit. If you don't want to pull in the dep for that that's fine by me

**Comment by maxice8** [comment]
I think it is better we just not enable jemalloc for now

**Comment by maxice8** [comment]
Pushed a fixed version of your commit:

- Removed patch that only apply to jeamalloc and was failing because patch-args is '-Np1' but the patch itself is meant for '-p0'

**Pull request closed without merging**